Minio/DataModel/BucketOperationsArgs.cs

```diff
@@ -187,12 +187,12 @@ public GetObjectListArgs WithListObjectsV1(bool useV1)
 
     internal override HttpRequestMessageBuilder BuildRequest(HttpRequestMessageBuilder requestMessageBuilder)
     {
-        // using System.Web; Not sure if we need to add query parameters like this vs. requestMessageBuilder.AddQueryParameter()
-        // var query = HttpUtility.ParseQueryString(string.Empty);
-        // query["foo"] = "bar<>&-baz";
-        // query["bar"] = "bazinga";
-        // string queryString = query.ToString();        {
-
+        var splitPrefixList = Prefix.Split('/');
+        if (Headers.ContainsKey("x-minio-extract"))
+            if ((splitPrefixList.Length == 1 && Prefix.EndsWith(".zip/")) ||
+                (splitPrefixList.Length > 1 && splitPrefixList[0].EndsWith(".zip")))
+                foreach (var h in Headers)
+                    requestMessageBuilder.AddOrUpdateHeaderParameter(h.Key, h.Value);
```

Comment:
Could you explain what this does?

Reply:
@klauspost 
Thanks for reviewing the code.

I checked the minio-go implementation by doing admin trace and noticed that the `"x-minio-exract"` header is included in the http signature only when the user provided `prefix` argument is set to `"<zip-file-name>.zip/"` or `<zip-file-name>.zip/<prefix1/../<prefixN>`.

Ideally, I should have been checking if the prefix starts with full zip file name instead of `".zip/"`, but I cannot get to the `objectName` information in this scope, hence ended up with this code.

I can either remove the above checks completely or  add a comment to explain why it is needed.

If we decide to remove the checks, the listing behavior is going to stay the same, but then we'll always have the `x-minio-extract` header included in the http signature, unlike minio-go implementation.